op-service/txmgr/metrics/tx_metrics.go

```diff
@@ -0,0 +1,32 @@
+package metrics
+
+import (
+	"github.com/ethereum-optimism/optimism/op-service/metrics"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type TxMetricer interface {
+	RecordL1GasFee(receipt *types.Receipt)
+}
+
+type TxMetrics struct {
+	TxL1GasFee prometheus.Gauge
+}
+
+var _ TxMetricer = (*TxMetrics)(nil)
+
+func MakeTxMetrics(ns string, factory metrics.Factory) TxMetrics {
+	return TxMetrics{
+		TxL1GasFee: factory.NewGauge(prometheus.GaugeOpts{
+			Namespace: ns,
+			Name:      "tx_l1gas_fee",
+			Help:      "",
+			Subsystem: "txmgr",
+		}),
+	}
+}
+
+func (t *TxMetrics) RecordL1GasFee(receipt *types.Receipt) {
+	t.TxL1GasFee.Set(float64(receipt.L1Fee.Uint64()))
```

Comment:
@trianglesphere @sebastianst  when I am running `devnet` `receipt.L1Fee` is always `nil`. Do you know why would the L1 fee would be `nil`?

Reply:
It's because this transaction manger is pointed at L1, but still uses op-geth which has additions to the receipt. The L1 Fee only makes sense in the context of a L2 transaction.